hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/clean/CleanPlanner.java

```diff
@@ -451,8 +451,12 @@ && noSubsequentReplaceCommit(earliestInstant.getTimestamp(), partitionPath)) {
    * <p>IMPORTANT: {@code fsView.getAllFileGroups} does not return pending file groups for metadata table,
    * file listing must be used instead.
    */
-  private boolean hasPendingFiles(String partitionPath) {
+  private boolean mayHavePendingFiles(String partitionPath) {
     try {
+      // As long as there are pending commits never delete empty partitions, because they may write files to any partitions.
+      if (!hoodieTable.getMetaClient().getCommitsTimeline().filterInflightsAndRequested().empty()) {
+        return true;
```

Comment:
For streaming ingestion, there should always be an pending instant on the timeline, and the following up logic may never reach.